numpy/core/function_base.py

```diff
@@ -1,11 +1,25 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
 
+def _index_deprecate(i, stacklevel=2):
+    try:
+        i = operator.index(i)
+    except TypeError:
+        msg = "%f cannot be safely interpreted as an integer." % i
```

Comment:
Thanks, maybe rather "object of type {} ...".format(type(i)) or just not %f but %s/{}. The object might also be a string in principle, and things would crash rather cryptically then. I admit it is very very unlikely, but since we may want to use this function elsewhere.